kernel/src/syscall/stat.rs

```diff
@@ -77,6 +81,60 @@ pub fn sys_fstatat(
     Ok(SyscallReturn::Return(0))
 }
 
+pub fn sys_statx(
+    dirfd: FileDesc,
+    filename_ptr: Vaddr,
+    flags: u32,
+    mask: u32,
+    statx_buf_ptr: Vaddr,
+    ctx: &Context,
+) -> Result<SyscallReturn> {
+    let user_space = ctx.user_space();
+    let filename = user_space.read_cstring(filename_ptr, MAX_FILENAME_LEN)?;
+    let flags =
+        StatxFlags::from_bits(flags).ok_or(Error::with_message(Errno::EINVAL, "invalid flags"))?;
```

Comment:
The error message "invalid statx flags" is more descriptive. "invalid flags" conveys basically nothing more than `EINVAL` and still confuses the user who might see it.